src/schedule/ll_schedule.c

```diff
@@ -190,15 +190,12 @@ static void schedule_ll_tasks_run(void *data)
 {
 	struct ll_schedule_data *sch = data;
 	struct ll_schedule_domain *domain = sch->domain;
-	uint32_t flags;
 
 	tr_dbg(&ll_tr, "timer interrupt on core %d, at %u, previous next_tick %u",
 	       cpu_get_id(),
 	       (unsigned int)platform_timer_get_atomic(timer_get()),
 	       (unsigned int)domain->next_tick);
 
-	irq_local_disable(flags);
```

Comment:
yes, holding interrupts blocked for the entire duration of all tasks is overly protective and should be relaxed. But to do that we have to understand well what possible races we might get. As far as I understand one of the possible preemption scenarios could be:
1. timer interrupt (core 0 only)
1.1. runs `schedule_ll_tasks_run()`
2. while interrupts are enabled an IPC message from the host can arrive, triggering an interrupt on core 0
2.1. the IPC interrupt will make the IPC EDF task runnable and perform a return from interrupt

after that I'm not sure what would happen. Which context would be swapped in? The timer IRQ, running the LL scheduler (sounds logical) or the IPC processing task `ipc_platform_do_cmd()`? Or is this even impossible because both the timer and the IPC interrupts are on level 2 and therefore cannot interrupt each other? If the only thing that can interrupt us here are higher level interrupts, then that's mostly just DW DMA interrupts, which are only used in DMA domain pipelines, e.g. with UP2 with PCM512x in master mode? I.e. we almost never can be interrupted here anyway?

Reply:
Yes, that's true, we don't have this context switching implementation yet, let me remove this change from the PR, thanks.